internal/fields/validate.go

```diff
@@ -23,17 +22,23 @@ import (
 
 // Validator is responsible for fields validation.
 type Validator struct {
-	schema []FieldDefinition
+	schema               []FieldDefinition
+	numericKeywordFields map[string]struct{}
 }
 
 // CreateValidatorForDataStream function creates a validator for the data stream.
-func CreateValidatorForDataStream(dataStreamRootPath string) (*Validator, error) {
+func CreateValidatorForDataStream(dataStreamRootPath string, numericKeywordFields []string) (*Validator, error) {
```

Comment:
What do you think about introducing `ValidatorOptions` that will contain `DataStreamRootPath string` and `NumericKeywordFields []string`?

I would be then:

```go
validator, _ := CreateValidatorForDataStream(ValidatorOptions{
  DataStreamRootPath: dsrp,
  NumericKeywordFields: nkf,
})

In case of further extensions we won't need to modify existing code. Another option is to refactor it to fluent interface (e.g. `WithDataStreamRootPath()`, `WithNumericKeywordFields()`).
```